src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -867,13 +771,15 @@ public void ReturnConnection(HttpConnection connection)
                     receivedUnexpectedData = connection.EnsureReadAheadAndPollRead();
                     if (!receivedUnexpectedData)
                     {
-                        ConnectionWaiter waiter = DequeueWaiter();
-                        waiter._cancellationTokenRegistration.Dispose();
-
-                        if (NetEventSource.IsEnabled) connection.Trace("Transferring connection returned to pool.");
-                        waiter.SetResult((connection, false, null));
-
-                        return;
+                        while (_waitersTail != null)
```

Comment:
Earlier we have a similar loop but using _waitersHead instead of _waitersTail.  Is there a reason for the difference, or could we be consistent?

Reply:
The only reason is that that's how the code was accessing these before. I agree, it would be better to be consistent here. I'll rework.